sources/packages/api/src-sql/UsersDraft/Create-UsersDraft.sql

```diff
@@ -0,0 +1,28 @@
+-- User Draft Table
+CREATE TABLE IF NOT EXISTS users_draft (
+  id SERIAL PRIMARY KEY,
+  -- Reference column
+  user_name VARCHAR(300) NOT NULL UNIQUE REFERENCES users(user_name) ON DELETE CASCADE,
```

Comment:
The student table is referencing the user through user_id, it would be good to keep the same reference field here. Is that any particular reason to create the relationship using the user_name?

Reply:
I welcome this comment, thanks for raising this. I did it because to check associated user we don't need to create any join query to fetch a draft for any user. Overall performance will be quick. And user_name is technically a business primary key of user table with index.

Reply:
I see. My point is that we should have this as a pattern and do it in all the places then.
I am not sure if we are not going to end up having the user id loaded from DB also if we are saving roles to our DB, for instance. Maybe we end up going to DB to load user roles and probably the user id, what would be available then to execute all the queries.